tests/mocha/blocks/procedures_test.js

```diff
@@ -1338,6 +1338,33 @@ suite('Procedures', function() {
           chai.assert.isTrue(
               callBlock2.disposed, 'Expected the second caller to be disposed');
         });
+
+    test('undoing and a procedure delete will still associate procedure ' +
+        'and caller with the same model',
+        function() {
+          const defBlock = createProcDefBlock(this.workspace);
+          createProcCallBlock(this.workspace);
+          // TODO: Apparently we need to call checkAndDelete to handle event
+          //   grouping, this seems like possibly a bug.
+          defBlock.checkAndDelete();
+          this.clock.runAll();
+
+          this.workspace.undo();
+          this.clock.runAll();
+
+          console.log(this.workspace.getTopBlocks());
+
+          const newDefBlock =
+              this.workspace.getBlocksByType('procedures_defnoreturn')[0];
+          const newCallBlock =
+              this.workspace.getBlocksByType('procedures_callnoreturn')[0];
+          console.log(newDefBlock, newCallBlock);
+
+          chai.assert.equal(
+              newDefBlock.getProcedureModel(),
+              newCallBlock.getProcedureModel(),
+              'Expected both new blocks to be associated with the same model');
```

Comment:
This checks whether the two blocks share the same model, but not whether the model is the same as the pre-deletion model. Do you need to check that as well?

Reply:
It won't be the same object as the previous model, because models get deleted when the definition blocks are deleted. But I added an assertion that the IDs match!